### PR TITLE
Add support for passing hash attributes

### PR DESCRIPTION
**Pull request by armiller**
This allows the ability to pass hashes through attributes to resources. I also went through and fixed all of the rubocop warnings.

### CODE REVIEW

**Comment by armiller** [comment]
@osuosl-cookbooks/chefs

**Comment by ramereth** [comment]
@armiller mind adding support for C7 in TK?

**Comment by armiller** [comment]
@ramereth updated.

File: libraries/resource_from_hash.rb [hunk 1]
```diff
@@ -1,52 +1,58 @@
...
+        attributes = indifferent_access(new_resource.hash, 'attributes').to_a
+        body = attributes.collect do |a|
+          "#{a.first}(#{a.last.inspect})"
+        end.join("\n")
+        suffix = '}'
+
+        # rubocop:disable Lint/UselessAssignment
+        block = eval(prefix + body + suffix)
+        name = indifferent_access(new_resource.hash, 'name').inspect
+        resource = string_symbol(new_resource.res.resource_name) + "(#{name},&block)"
```
Comment (ramereth) [inline libraries/resource_from_hash.rb]: libraries/resource_from_hash.rb:52:81: C: Line is too long. [85/80]

Comment (armiller) [inline libraries/resource_from_hash.rb]: sigh. vim's linter stopped working

**Comment by ramereth** [comment]
:+1:

**Pull request merged**